### ISSUE TITLE
setting to extend sudo password validity

### ISSUE THREAD

**Issue by teto** [NixOS/nixpkgs#40157]
## Issue description
I find myself typing my sudo password way too frequently. I wish I could set the timer to extend the period during which "sudo" works without password input.

## Technical details

on nixos-unstable

**Comment by jtojnar** [comment]
Perhaps something like the following:

```
  security.sudo.extraConfig = ''
    Defaults        timestamp_timeout=30
  '';
```

See `CONFIGURATION.NIX(5)` and `SUDOERS(5)`.

**Comment by teto** [comment]
Sounds good, I feel like it would be a setting worth adding to nixos (rather than through extraConfig) but feel free to close the issue if that's not the case. I might even submit the PR if needed.

**Comment by xeji** [comment]
I don't think we should create an extra nixos option for this. It is related to other options, like `timestamp_type`, `timestampdir`, `timestampowner`.  IMO it's better to provide a safe default only and let users configure these advanced settings manually in `extraConfig`.

**Comment by poiNt3D** [comment]
1. It's unnecessarily short
2. It encourages use of simple passwords
